resources/fetchersManager/factory/factory.go

```diff
@@ -0,0 +1,94 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package factory
+
+import (
+	"context"
+	"fmt"
+
+	awssdk "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/elastic/beats/v7/x-pack/libbeat/common/aws"
+	"github.com/elastic/cloudbeat/config"
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/cloudbeat/resources/providers/awslib"
+	"github.com/elastic/cloudbeat/uniqueness"
+	"github.com/elastic/elastic-agent-libs/logp"
+	k8s "k8s.io/client-go/kubernetes"
+)
+
+type RegisteredFetcher struct {
+	Fetcher   fetching.Fetcher
+	Condition []fetching.Condition
+}
+
+type FetchersMap map[string]RegisteredFetcher
+
+// NewFactory creates a new factory based on the benchmark name
+func NewFactory(
+	ctx context.Context,
+	log *logp.Logger,
+	cfg *config.Config,
+	ch chan fetching.ResourceInfo,
+	le uniqueness.Manager,
+	k8sClient k8s.Interface,
+	identityProvider func(cfg awssdk.Config) awslib.IdentityProviderGetter,
+	awsConfigProvider awslib.ConfigProviderAPI,
+) (FetchersMap, error) {
+	if cfg == nil {
+		return nil, fmt.Errorf("config is nil")
+	}
+
```

Comment:
```suggestion
```

I wonder if this kind of defensive programming is needed that deep in the code, if we do it consistently it will lead to a lot of bloat. The caller should be responsible in passing a valid `cfg` at this point or throw an error.